opendcs-rest-api/src/test/java/org/opendcs/odcsapi/res/PlatformResourcesTest.java

```diff
@@ -0,0 +1,427 @@
+	{
+		assertEquals(apiPlatformSensors.size(), iterSize(platformSensors));
+		int i = 0;
+		while (platformSensors.hasNext())
+		{
+			PlatformSensor platformSensor = platformSensors.next();
+			ApiPlatformSensor apiPlatformSensor = apiPlatformSensors.get(i);
+
+			assertEquals(platformSensor.sensorNumber, apiPlatformSensor.getSensorNum());
+			assertEquals(platformSensor.site.getId().getValue(), apiPlatformSensor.getActualSiteId());
+			assertEquals(platformSensor.getProperties(), apiPlatformSensor.getSensorProps());
+			assertEquals(platformSensor.getUsgsDdno(), apiPlatformSensor.getUsgsDdno());
+
+			i++;
+		}
+	}
+
+	private static void assertMatchMedium(ArrayList<ApiTransportMedium> apiTransportMedium, Iterator<TransportMedium> transportMedium)
+	{
+		assertEquals(apiTransportMedium.size(), iterSize(transportMedium));
+		int i = 0;
+		while (transportMedium.hasNext())
+		{
+			TransportMedium tm = transportMedium.next();
+			ApiTransportMedium atm = apiTransportMedium.get(i);
+
+			assertEquals(tm.getMediumType(), atm.getMediumType());
+			assertEquals(tm.getMediumId(), atm.getMediumId());
+			assertEquals(tm.scriptName, atm.getScriptName());
+			assertEquals(tm.channelNum, atm.getChannelNum());
+			assertEquals(tm.assignedTime, atm.getAssignedTime());
+			assertEquals(tm.transmitWindow, atm.getTransportWindow());
+			assertEquals(tm.transmitInterval, atm.getTransportInterval());
+			assertEquals(tm.getTimeZone(), atm.getTimezone());
+			assertEquals(tm.getBaud(), atm.getBaud());
+			assertEquals(tm.getStopBits(), atm.getStopBits());
+			assertEquals(tm.getDataBits(), atm.getDataBits());
+			assertEquals(tm.getParity(), atm.getParity().charAt(0));
+			assertEquals(tm.getUsername(), atm.getUsername());
+			assertEquals(tm.getPassword(), atm.getPassword());
+			assertEquals(tm.isDoLogin(), atm.getDoLogin());
+		}
+	}
+
+	@Test
+	void testPlatformStatusMap()
+	{
+		PlatformStatus status = new PlatformStatus(DbKey.createDbKey(9987500L));
+		status.setDesignator("Platform Designator");
+		status.setAnnotation("Platform Annotation");
+		status.setChecked(true);
+		status.setLastContactTime(Date.from(Instant.parse("2021-07-02T12:00:00Z")));
+		status.setSiteName("Platform location");
+		status.setLastFailureCodes("System Failure 2");
+		status.setLastErrorTime(Date.from(Instant.parse("2021-07-01T12:00:00Z")));
+
+		ArrayList<ApiPlatformStatus> apiStatus = map(status);
+		assertNotNull(apiStatus);
+		assertFalse(apiStatus.isEmpty());
+		assertEquals(1, apiStatus.size());
+		ApiPlatformStatus apiStat = apiStatus.get(0);
+		assertEquals(status.getAnnotation(), apiStat.getAnnotation());
+		assertEquals(status.getLastContactTime(), apiStat.getLastContact());
+		assertEquals(status.getPlatformId().getValue(), apiStat.getPlatformId());
+		assertEquals(status.getLastRoutingSpecName(), apiStat.getRoutingSpecName());
+	}
+
+	private static int iterSize(Iterator<?> it)
```

Comment:
interesting.  
+1 b/c its clear what this is doing but -1 b/c its weird to have to do this.

Reply:
Yeah, if the implementation gave us more than just the iterator this wouldn't be needed.

Reply:
These will be migrated to a more standard DAO style, but platforms is not first on the list (Enum is in the works, routingspecs/schedule entries are probably next) so for now you're stuck with doing this.